### PR TITLE
Feat(eos_cli_config_gen): Create maintenance configuration

### LINKED ISSUE #922: Support for maintenance mode profile configuration
**Issue Type**

- [ ] Template enhancement
- [x] Role enhancement
- [ ] Documentation enhancement (avd.sh)

**Is your feature request related to a problem? Please describe.**

We have to adjust the maintenance mode profile configuration on all L3/MLAG-Leaf switches to make maintenance mode work in our environment.

**Describe the solution you'd like**

We have to configure a maintenance profile interface with certain values like the following one and make it the default.

```
maintenance
   profile interface int_maint default
   !
   profile interface int_maint
      rate-monitoring load-interval 100
      rate-monitoring threshold 1000
      shutdown max-delay 100
```

There is currently no template available within the eos_cli_config_gen role.
The data model could perhaps look like this:
```
maintenance:
  profile:
    interface:
      < profile_interface_1 >:
        rate_monitoring:
          load_interval: < seconds >
          threshold: < kbps >
        shutdown:
          max_delay: < seconds >     
        default: < true | false >
```

**Describe alternatives you've considered**

The current alternative is, to create a config_let within CVP and bind it to the containers of our leafs manually within CVP or "automated" by adding the config_let via AVD.
Our goal is that Ansible code is the single source of configuration.  

**Additional context**

### PR DESCRIPTION
**Pull request by daanvdsanden**
## Change Summary

This PR introduces the possibility to create configuration for the Arista EOS maintenance stanza
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [X] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #922 

## Component(s) name

`arista.avd.cli_config_gen`

## Proposed changes
Added the psobility to create maintenance configuration stanze

## How to test
Ran the molecule suite to create configuration.
Afterwardes verified the generated config.

## Checklist:

- [X] My code has been rebased from devel before I start
- [X] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contributing/) document.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [ ] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

**Comment by ClausHolbechArista** [comment]
Please remove the implementation of `group` from this PR, so it is purely `maintenance` that is covered. We have decided to go with the implementation in #1017 for `group`. I will try to align the two PRs during review process, so we minimize any conflicts.
It is important that your documentation template only covers the `maintenance` data model. We will include the two areas as separate documentation templates like:

```
# Maintenance Mode
## Maintenance            <-- documentation template from this PR
## Group                  <-- documentation template from #1017 
```

**Comment by daanvdsanden** [comment]
Hi besides that it my commit I feel the data model in #1017 is not consistent with the rest of the models. I expect that the following:
```yaml
groups:
  interfaces:
    <interface_1>:
      <body>
```
Translates to 
```
!
groups
   interfaces
     interface_1
       body
!
```
If we compare the output we need (with the interface and bgp groups all in the root of the config) and compare it with routing, we have ```router_bgp```, ```router_ospf```, etc. and not:
```yaml
routing:
  bgp:
    <bgp_config>
  ospf:
    <ospf_config>
```
For interfaces we also have ```ethernet_interface```, ```port_channel_interfaces```, ```vlan_interfaces```, etc and not:
```yaml
interfaces:
  ethernet:
    <interface_1>:
  port_channel:
    <interface_2>:
```
So if this logic is mirrored to the groups I would expect:
```yaml
bgp_groups_:
  < bgp_group_1 >:
    vrf: < vrf_name >
    neighbors:
    - < neighbor_1 >
    - < neighbor_2 >
    - < peer_group >
    maintenance:
      bgp_profiles:
      - < bgp_profile_1 >
interface_groups:
  < interface_group_1 >:
    interfaces:
      ethernet: < ethernet_interface_range >
      port_channel: < port_channel_interface_range >
      vlan: < vlan_interface_range >
    maintenance:
      bgp_profiles:
      - < bgp_profile_1 >
      interface_profiles:
      - < interface_profile_1 >
      - < interface_profile_2 >
```
And not the model proposed in #1017

**Comment by carlbuchmann** [comment]
@daanvdsanden - Thank you for your continued input on the PR and data models.

This is indeed a very tricky one, and I was of the same mindset when I originally reviewed this myself, and after discussing with both @ClausHolbechArista and @aphillipps we agreed to continue with the data model proposal from @c-po in PR #1017.

The challenge with this command is that possible inputs differ from the generated outputs. 
From a config standpoint the following configuration is accepted:

```shell
group interface test
   interface Ethernet 1
   interface Ethernet 3-4
   interface vlan 3
   interface vlan 5-6
   interface vlan 7
   interface Port-Channel 1,10-12
```

and results in:

```shell
group interface test
   interface Et1,3-4
   interface Po1,10-12
   interface Vl3,5-7
   exit
```

In our opinion, in this case, the input is much more readable and more flexible to work with than the resulting configuration from show running-config.

Having a more flexible input, and still accepted in a config session will make it easier in the long run when build capabilities in the abstraction role `eos_designs` to implement best practices here.

**Comment by daanvdsanden** [comment]
I updated the code to update the datamodel to, which I can also to generate config that is equal to the running config: 
```yaml
bgp_groups_:
  < bgp_group_1 >:
    vrf: < vrf_name >
    neighbors:
    - < neighbor_1 >
    - < neighbor_2 >
    - < peer_group >
    maintenance:
      bgp_profiles:
      - < bgp_profile_1 >
interface_groups:
  < interface_group_1 >:
    interfaces:
    - < interface_range_1 >
    - < interface_range_2>
    maintenance:
      bgp_profiles:
      - < bgp_profile_1 >
      interface_profiles:
      - < interface_profile_1 >
      - < interface_profile_2 >
```
I still think the group definition should be flattened same as is done with routing and interfaces.

**Comment by ClausHolbechArista** [comment]
@daanvdsanden Thank you for your commitment to making the data-model consistent. We _are_ trying to mimic the EOS CLI structure as much as possible, but also to strike a balance with readability of the data model etc. So you will see variances throughout the data models depending on the area.
I actually agree on your comments here, and I am working on implementing a flattened model into pr #1017.
Again, please remove the groups coverage from this PR, so we can work on each area separately.

FYI This is the data model I plan to implement in #1017 (not approved by other maintainers yet!):

### Maintenance Mode

#### BGP Groups

```yaml
bgp_groups:
  < group_name >:
    vrf: "< vrf_name >"
    neighbors:
      - "< ip_address >"
      - "< ipv6_address >"
      - "< peer_group_name >"
    bgp_maintenance_profiles:
      - < profile_name >
```

#### Interface Groups

```yaml
interface_groups:
  < group_name >:
    interfaces:
      - "< interface_or_interface_range >"
    bgp_maintenance_profiles:
      - "< profile_name >"
    interface_maintenance_profiles:
      - "< profile_name >"
```

**Comment by carlbuchmann** [comment]
@daanvdsanden - thank you for your patience, PR #1017 has now been merged! Please rebase with the latest devel branch. Please let us know if you need any assistance with addressing the conflicts!

**Comment by daanvdsanden** [comment]
@carlbuchmann All is cleaned up and merged without any conflicts left.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/maintenance-mode.j2
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/maintenance-mode.j2]: Please adjust the comment here since now you are implementing this.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/maintenance.j2
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/maintenance.j2]: missing variable check for `maintenance.bgp_profiles[bgp_profile].initiator.route_map_inout`

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/maintenance.j2
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/maintenance.j2]: missing trailing `|`. Also please adjust the next line to fit the width of headings. Same for other tables. This eases readability.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/maintenance.j2
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/maintenance.j2]: to make this more readable, please set some temporary variables like `set row_load_interval = maintenance.interface_profiles[interface_profile].rate_monitoring.load_interval | arista.avd.default(60)` and use those shorter variables in the table fields, which can then stay on a single line. Same for "Maintenance Units" table.
Also please add a unit to the heading. Like `(s)` for interval and `(kbps)` for threshold to help the reader.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/maintenance.j2
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/maintenance.j2]: Please ensure consistent 1 space before and after the table values. Same for other tables.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/maintenance.j2
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/maintenance.j2]: Our `arista.avd.default` filter can take multiple values, so this can be simplified to `maintenance.units[unit].profile | arista.avd.default(maintenance.default_unit_profile, 'Default')`

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/maintenance.j2
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/maintenance.j2:66]: please render `no quiesce` if specifically set to false.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: This level is no longer needed after you convinced us of using `bgp_groups` and `interface_groups`. So I suggest that we flatten it a bit by removing this level.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/maintenance.j2
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/maintenance.j2:2]: Impressive that you have caught all EOS behaviors for setting `!` in here. It makes it a bit cluttered, so I will ask @carlbuchmann and @titom73 for their view on this.

**Review by ClausHolbechArista — DISMISSED** [review]
I might need another review round when these things have been updated, but it is looking good so far. Thank you for this contribution and your persistence on consistent data models.

**Comment by daanvdsanden** [comment]
I updated the code. I had some problems with my git commands (which I think I resolved, but it resulted in some extra tags on the pull request, sorry for that). Let me know if the explanation marks need to go or not...

**Review by carlbuchmann — APPROVED** [review]
LGTM, please resolve the conflict and we will merge!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/maintenance.j2
Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/maintenance.j2:2]: We will keep current template behavior.

**Review by carlbuchmann — APPROVED** [review]
LGTM! Thanks again for this PR!

**Pull request merged**
